### DISCUSSION TITLE
Paint gun hud ideas

### DISCUSSION THREAD

**Discussion by JJL772** [StrataSource/Portal-2-Community-Edition#706] [Ideas]
We'll be designing a brand new paint gun hud (the old one doesn't even work), so let's get some ideas brainstormed! None of these images are "final"- these are very rough drafts made in paint/inkscape.

Relevant issue and previous discussion here: #705 

## Paint select HUD

For the paint selector HUD element, we have the following concepts:

---
![image](https://user-images.githubusercontent.com/30549015/130755320-c1ec2c3f-c683-4102-906d-4584ef6a58ce.png)

---
![image](https://user-images.githubusercontent.com/19717056/130736764-d0f5a53d-cdbd-41d8-81c4-1ab0b83a6859.png)

## Paint gun crosshair

For the paint gun's crosshair (which should definitely be unique from the portal gun's) we have the following concepts:

---
![image](https://user-images.githubusercontent.com/19717056/130736977-2d15d3f8-c89d-4874-a17f-9fb61c381c6d.png)

---
![image](https://user-images.githubusercontent.com/19717056/130737074-ebbb6c47-964a-435b-bb1d-a4d43a0cc6e7.png)

**Comment by Critfish** [comment]
![image](https://user-images.githubusercontent.com/58682909/130739223-aab8ed62-5561-4211-a7ab-0e24c5478279.png)
How about a "raindrop" shape?
edit: made it a bit better

**Comment by Konclan** [comment]
**Paint Selection Radial Menu**

Here's my contribution:
Its designed like the coop ping menu but instead of the middle being cancel it defaults to swap to next paint. You can move your mouse to either select a new paint or select the paint you already have to cancel. This menu merges two functions of cycling paint and quick selection. The radial menu will expand and contract based on what paints the player has available.

Here is a proof of concept I made in paint. Its not great visually but hopefully it demonstrates my idea clearly.
![Paint Gun GUI Concept](https://user-images.githubusercontent.com/16567080/130745708-4d2fcd47-9ed0-4a10-ba80-92ccc089ba23.png)

**Reply by Hambrix** [reply]
![hud2](https://user-images.githubusercontent.com/84916624/130836637-3bb266dd-26c5-427e-a6b9-bc4a8428fef7.jpg)

**Reply by Delikatessen** [reply]
I like this one, since the Paint Gun is from Old Aperture lore-wise, it would be nice to have the GUI designed with that retro style in mind. Maybe having it switch from retro to modern when you go from the Paint Gun to Portal Gun? Might be too much, but sounds cool.

**Reply by vrad-exe** [reply]
The radial menu is a neat idea, it solves the problem of how to conveniently select between a possibly infinite number of custom gels. We're still not sure about the crosshair though.

**Reply by ooooggll** [reply]
I feel like just having the portal gun crosshair for the paint gun would be good enough. Obviously instead of blue and orange it would be whatever gel you currently have selected. Or instead of having it "tilted" like the portal gun crosshair you could have it evenly half and half.
![paintgun_crosshair](https://user-images.githubusercontent.com/99767881/154366640-83e51efa-27ab-4fb6-81cc-5a6ab0da00e4.png)
Something like this for that second concept. But honestly I think the default portalgun crosshair would work fine.
Obviously it doesn't look very neat in the picture but it's just a basic idea.

**Reply by ooooggll** [reply]
Also I don't know what right clicking with the paint gun currently does, if anything, but I have a couple ideas for what the right click function could be:
1. Shoots water (to cleanse gels)
2. Opens the radial gel selection menu (that Konclan suggested)
3. When selecting gels, you can choose a primary and secondary gel, so if you know you'll be using repulsion and adhesion for instance, you don't have to switch as often. So right clicking would shoot the "secondary" gel.

I think the option that probably makes the most sense would be for it to open the radial menu but idk. I think all would be cool, or if anyone else has any ideas I'd like to hear them.

**Reply by FrozenDragon0** [reply]
Right clicking just shoots out cleansing gel at the moment.

**Reply by ooooggll** [reply]
Well cool, that makes sense I suppose.

**Comment by slambo311** [comment]
With all these ideas, why not just have a setting to switch between them?

**Reply by AWildErin** [reply]
As somebody who would probably end up having to implement this, I’d rather just have one, not all.

**Reply by craftablescience** [reply]
Seconded, if you want to use a different one in a mod you can edit the panorama files

**Reply by vrad-exe** [reply]
Yep, the nice thing about Panorama is that it's totally customizable, you can add any of these in place of the default one or make something entirely new.
